advanced_topics.qmd

```diff
@@ -266,15 +277,371 @@ pums_person |>
   collect()
 ```
 
+### Using UDFs for regression model predictions
+
+Another example where UDFs can be useful is where you've created a model based on a subset of your data and you want to use it to make predictions.
+Let's say we've created a linear model, using education and employment status to predict hours worked per week for respondents in 2018.
+When creating this model we aren't using the standard `stats::lm()` because our data is weighted. 
+The `survey` package has an equivalent regression function `survey::svyglm()` which applies the weights in the weight column during the regression. 
+This isn't strictly necesary for the demonstration here, but it does show how UDFs work not just with base R functions or functions in the tidyverse, but any R code that you come up with.
+
+```{r}
+#| label: model-commute-time
+model_data <- open_dataset("data/person/") |>
+    filter(year == 2018) |>
+    filter(AGEP >= 16, AGEP < 65, !is.na(COW)) |>
+    transmute(
+      self_employed = str_detect(as.character(COW), "Self-employed"),
+      edu = case_when(
+        grepl("bachelor|associate|professional|doctorate|master", SCHL, TRUE) ~
+          "College or higher",
+        grepl("college", SCHL, TRUE) ~
+          "Some college",
+        grepl("High school graduate|high school diploma|GED", SCHL, TRUE) ~
+          "High school",
+        TRUE ~ "Less than High school"
+      ),
+      hours_worked = WKHP,
+      PWGTP = PWGTP
+    ) |>
+    collect()
+
+hours_worked_model <- survey::svyglm(
+  hours_worked ~ self_employed * edu,
+  design = survey::svydesign(ids = ~1, weights = ~PWGTP, data = model_data)
+)
+summary(hours_worked_model)
+```
+
+We can use this model to predict what our model would have on this data.
```

Comment:
This sentence doesn't quite scan for me, but wasn't sure what to change it to sorry

Reply:
Aaaah that's because it's half complete. I added the bit I had in my head but didn't actually write out. I also rearranged and expanded a little bit more detail around there too.